### PR TITLE
Fix aws_db_instance false-positives

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #542 
| ❓ Documentation  | no

## Description

Fix false positive for aws_db_instance CharacterSetName
Add Acc test.
Maybe it's need to change pkg/iac/terraform/state/test/db_instance/terraform.tfstate and pkg/iac/terraform/state/test/db_instance/result.golden.json, let me know.

### CODE REVIEW

File: pkg/resource/aws/aws_db_instance_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		Paths: []string{"./testdata/acc/aws_db_instance"},
+		Args:  []string{"scan", "--filter", "Type=='aws_db_instance'"},
+		Checks: []acceptance.AccCheck{
+			{
+				Check: func(result *test.ScanResult, stdout string, err error) {
+					if err != nil {
+						t.Fatal(err)
+					}
+					result.AssertDriftCountTotal(0)
+					result.Equal(1, result.Summary().TotalManaged)
```
Comment (wbeuil) [inline pkg/resource/aws/aws_db_instance_test.go:21]: You could use:

```
result.AssertInfrastructureIsInSync()
result.AssertManagedCount(1)
```

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by sundowndev — DISMISSED** [review]
great job 👍🏼

File: pkg/resource/aws/aws_db_instance_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package aws_test
+
+import (
+	"testing"
+
+	"github.com/cloudskiff/driftctl/test"
+	"github.com/cloudskiff/driftctl/test/acceptance"
+)
+
+func TestAcc_AwsDbInstance_WithCharacterSetName(t *testing.T) {
```
Comment (wbeuil) [inline pkg/resource/aws/aws_db_instance_test.go:10]: Can you remove the `_WithCharacterSetName`?

File: pkg/resource/aws/aws_db_instance_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/cloudskiff/driftctl/test/acceptance"
+)
+
+func TestAcc_AwsDbInstance(t *testing.T) {
+	acceptance.Run(t, acceptance.AccTestCase{
+		Paths: []string{"./testdata/acc/aws_db_instance"},
+		Args:  []string{"scan", "--filter", "Type=='aws_db_instance'"},
+		Checks: []acceptance.AccCheck{
+			{
+				Check: func(result *test.ScanResult, stdout string, err error) {
```
Comment (wbeuil) [inline pkg/resource/aws/aws_db_instance_test.go:16]: Can you add the ENV parameter also:

```
Env: map[string]string{
					"AWS_REGION": "us-east-1",
				},
```

File: pkg/resource/aws/testdata/acc/aws_db_instance/db_instance.tf [hunk 4]
```diff
@@ -0,0 +1,11 @@
+resource "aws_db_instance" "default" {
```
Comment (wbeuil) [inline pkg/resource/aws/testdata/acc/aws_db_instance/db_instance.tf:1]: At the top of this file

```
provider "aws" {
  region = "us-east-1"
}

terraform {
  required_providers {
    aws = "3.19.0"
  }
}
```

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by wbeuil — APPROVED** [review]

File: pkg/resource/aws/aws_db_instance_ext.go [hunk 5]
```diff
@@ -14,9 +14,15 @@ func (r *AwsDbInstance) NormalizeForState() (resource.Resource, error) {
 	if r.ApplyImmediately != nil && !*r.ApplyImmediately {
 		r.ApplyImmediately = nil
 	}
+	if r.CharacterSetName != nil && *r.CharacterSetName == "" {
```
Comment (eliecharra) [inline pkg/resource/aws/aws_db_instance_ext.go:17]: This fix need to be applied to the new infrastructure design in main branch @lotoussa @wbeuil

Comment (wbeuil) [reply pkg/resource/aws/aws_db_instance_ext.go:17]: Good catch will do it now !

Comment (wbeuil) [reply pkg/resource/aws/aws_db_instance_ext.go:17]: Done d427632

**Pull request merged**